tests/auto/jobs/build.py

```diff
@@ -152,68 +198,101 @@ def clone_pr_repo(job_obj, workdir):
+        return not build_failed
     else:
         logger.critical(f'Could not find {job_obj.machine}'
                         f'.{job_obj.compiler} '
                         f'{job_obj.preq_dict["action"]} log')
         raise FileNotFoundError
+
+
+def process_gen(job_obj, gen_log_loc, gen_log_name):
+    """
+    Runs after a rocoto workflow has been generated
+    Checks to see if an error has occurred
+    """
+    logger = logging.getLogger('BUILD/PROCESS_GEN')
+    gen_log = f'{gen_log_loc}/{gen_log_name}'
+    error_string = 'ERROR'
+    gen_failed = False
+    if os.path.exists(gen_log):
+        with open(gen_log) as fname:
+            for line in fname:
+                if error_string in line:
+                    job_obj.comment_append('Generating Workflow Failed')
+                    gen_failed = True
+                    logger.info('Generating workflow failed')
+                if gen_failed:
+                    job_obj.comment_append(f'{line.rstrip()}')
+
+
+def process_expt(job_obj, expt_dirs):
+    """
+    Runs after a rocoto workflow has been started to run one or more expts
+    Assumes that more expt directories can appear after this job has started
+    Checks for success or failure for each expt
+    """
+    logger = logging.getLogger('BUILD/PROCESS_EXPT')
+    expt_done = 0
+    repeat_count = 72
+    complete_expts = []
+    expt_list = os.listdir(expt_dirs)
+    complete_string = "This cycle is complete"
+    failed_string = "FAILED"
+
+    while (expt_done < len(expt_list)) and repeat_count > 0:
+        time.sleep(300)
+        repeat_count = repeat_count - 1
+        expt_list = os.listdir(expt_dirs)
+        logger.info('Experiment dir after return of end_to_end')
+        logger.info(expt_list)
+        for expt in expt_list:
+            expt_log = expt_dirs + '/' + expt + \
+                '/log/FV3LAM_wflow.log'
+            if os.path.exists(expt_log) and expt not in complete_expts:
+                with open(expt_log) as fname:
+                    for line in fname:
+                        if complete_string in line:
+                            expt_done = expt_done + 1
+                            job_obj.comment_append(f'Experiment done: {expt}')
+                            job_obj.comment_append(f'{line.rstrip()}')
+                            logger.info(f'Experiment done: {expt}')
+                            complete_expts.append(expt)
+                        else:
+                            if failed_string in line:
+                                expt_done = expt_done + 1
+                                job_obj.comment_append('Experiment failed: '
+                                                       f'{expt}')
+                                job_obj.comment_append(f'{line.rstrip()}')
+                                logger.info(f'Experiment failed: {expt}')
+                                complete_expts.append(expt)
+    logger.info(f'Wait Cycles completed: {72 - repeat_count}')
```

Comment:
Here's 72 again...maybe this could be saved in a variable to reduce duplication?

Reply:
I am now saving it in a variable.

I tested some individual lines before making the changes. It's running on Hera now. I'm just going to commit and push the code so you can see it, and then I'll do it again later if anything needs to be fixed. These are pretty small changes.